pkg/cmd/server/api/validation/validation.go

```diff
@@ -224,3 +225,51 @@ func ValidateExtendedArguments(config api.ExtendedArguments, flagFunc func(*pfla
 
 	return allErrs
 }
+
+func ValidateLDAPSyncConfig(config api.LDAPSyncConfig) fielderrors.ValidationErrorList {
+	var errors fielderrors.ValidationErrorList
+	parsedURL, err := url.Parse(config.URL)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("given LDAP url invalid: %v", err))
+	}
+
+	scheme, err := ldaputil.DetermineLDAPScheme(parsedURL.Scheme)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	if scheme == ldaputil.SchemeLDAPS && config.Insecure {
+		errors = append(errors, fmt.Errorf("TLSInsecureSkipVerify cannot be used with an LDAPS connnection"))
+	}
+
+	_, err = util.CertPoolFromFile(config.CA)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("error loading cert pool from CA file %s: %v", config.CA, err))
+	}
+
+	if len(config.Attributes.Name) == 0 {
+		errors = append(errors, fmt.Errorf("which LDAP user attribute maps to an OpenShift User's name must be given"))
+	}
+
+	if config.UserDefinedGroupNameMapping {
+		if len(config.LDAPGroupUIDToOpenShiftNameMapping) == 0 {
+			errors = append(errors, fmt.Errorf("user-defined LDAP group unique id mapping chosen, but no mapping given"))
+		}
+	}
+
+	templates := []api.LDAPQueryTemplate{config.GroupQueryTemplate, config.UserQueryTemplate}
```

Comment:
create a method to check `LDAPQueryTemplate` types and then use `.Prefix()` to get the field names correct.